giskard_vision/landmark_detection/detectors/base.py

```diff
@@ -0,0 +1,112 @@
+import os
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Sequence
+
+import cv2
+
+from giskard_vision.detectors.base import DetectorVisionBase, IssueGroup, ScanResult
+from giskard_vision.landmark_detection.tests.base import TestDiff
+from giskard_vision.landmark_detection.tests.performance import NMEMean
+from giskard_vision.utils.errors import GiskardImportError
+
+Cropping = IssueGroup(
+    "Cropping", description="Cropping involves evaluating the landmark detection model on specific face areas."
+)
+
+Ethical = IssueGroup(
+    "Ethical",
+    description="The data are filtered by ethnicity to detect ethical biases in the landmark detection model.",
+)
+
+Pose = IssueGroup(
+    "Head Pose",
+    description="The data are filtered by head pose to detect biases in the landmark detection model.",
+)
+
+Robustness = IssueGroup(
+    "Robustness",
+    description="Images from the dataset are blurred, recolored and resized to test the robustness of the model to transformations.",
+)
+
+
+class LandmarkDetectionBaseDetector(DetectorVisionBase):
+    """
+    Abstract class for Landmark Detection Detectors
+
+    Methods:
+        get_dataloaders(dataset: Any) -> Sequence[Any]:
+            Abstract method that returns a list of dataloaders corresponding to
+            slices or transformations
+
+        get_results(model: Any, dataset: Any) -> Sequence[ScanResult]:
+            Returns a list of ScanResult containing the evaluation results
+
+        get_scan_result(self, test_result) -> ScanResult:
+            Convert TestResult to ScanResult
+    """
+
+    @abstractmethod
+    def get_dataloaders(self, dataset: Any) -> Sequence[Any]:
+        ...
+
+    def get_results(self, model: Any, dataset: Any) -> Sequence[ScanResult]:
+        dataloaders = self.get_dataloaders(dataset)
+
+        results = []
+        for dl in dataloaders:
+            test_result = TestDiff(metric=NMEMean, threshold=1).run(
+                model=model,
+                dataloader=dl,
+                dataloader_ref=dataset,
+            )
+
+            # Save example images from dataloader and dataset
+            current_path = str(Path())
+            os.makedirs(f"{current_path}/examples_images", exist_ok=True)
+            filename_examples = []
+
+            if hasattr(test_result, "indexes_examples") and test_result.indexes_examples is not None:
+                index_worst = test_result.indexes_examples[0]
+            else:
+                index_worst = 0
+
+            if dl.dataloader_type != "filter":
```

Comment:
remove `dataloader_type` completely, and check `if isinstance(dl, FilteredDataLoader)`